### PR TITLE
Fix LOP and Unoswap bugs

### PR DESCRIPTION
**Pull request by fatayri**
Brief comments on the purpose of your changes:

In the previous PR there were two bugs:

1. Unoswap trades had wrong datatypes in the pools columns
2. Maker and Taker amounts were in the wrong columns

Fixed both

### CODE REVIEW

**Comment by jeff-dude** [comment]
@masquot are you working on this? i merged the old PR to reflect the current state of caladan, even with usd amounts amplified.

**Comment by jeff-dude** [comment]
fyi @augustog on 1inch v1 engine

File: deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql [hunk 1]
```diff
@@ -260,38 +266,50 @@ WITH rows AS (
...
             union all
-            select "output_returnAmount", "amount", "srcToken", "pools", "call_tx_hash", "call_trace_address", "call_block_time", "contract_address" 
-            from oneinch_v4."AggregationRouterV4_call_unoswapWithPermit"  
+            select "output_returnAmount", "amount", "srcToken", "pools", "call_tx_hash", "call_trace_address", "call_block_time", "contract_address"
+            from oneinch_v4."AggregationRouterV4_call_unoswapWithPermit"
+            where call_success
+                and call_block_time >= start_ts
+                and call_block_time < end_ts
+            union all
+            select "output_returnAmount", "amount", "srcToken", numeric2bytea(pools) as pools, "call_tx_hash", "call_trace_address", "call_block_time", "contract_address"
```
Comment (masquot) [inline deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql:293]: Thank you @fatayri. The code fails during runtime on ` numeric2bytea(pools)` with the following error:

> No function matches the given name and argument types. You might need to add explicit type casts.

File: deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql [hunk 2]
```diff
@@ -260,38 +266,50 @@ WITH rows AS (
...
+            from oneinch."AggregationRouterV5_call_unoswapToWithPermit"
+            where call_success
+                and call_block_time >= start_ts
+                and call_block_time < end_ts
+            union all
+            select "output_returnAmount", "amount", "srcToken", numeric2bytea(pools) as pools, "call_tx_hash", "call_trace_address", "call_block_time", "contract_address"
```
Comment (masquot) [inline deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql:299]: See comment above

**Review by masquot — CHANGES_REQUESTED** [review]

File: deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql (hunk 1 above)
Comment (masquot) [reply deprecated-dune-v1-abstractions/ethereum/dex/trades/insert_1inch.sql:293]: See also https://dune.com/queries/1714974:

> Error: Function numeric2bytea(numeric[]) does not exist at line 5, position 3.

**Comment by bh2smith** [comment]
Tagging @fleupold who is also interested in the progress of this.

**Comment by bh2smith** [comment]
Note also that this PR closes https://github.com/duneanalytics/spellbook/issues/2197

**Comment by gentrexha** [comment]
Hey everyone, do you guys have an estimate when this will be fixed?

**Comment by fatayri** [comment]
Just pushed a fix, sorry for the delay 

@gentrexha @bh2smith

**Comment by bh2smith** [comment]
Now we just gotta get it merged before v1 is over! Wen V2?

**Comment by jacubcub** [comment]
@masquot can this be merged now?

**Comment by jeff-dude** [comment]
@dot2dotseurat fyi on v1 PR

**Comment by dot2dotseurat** [comment]
Hi, sorry for the inconvenience but all abstractions in the deprecated-dune-v1-abstractions have been migrated to https://github.com/duneanalytics/dune-v1-abstractions. 

Can these changes be re-opened as a PR there?

**Comment by jacubcub** [comment]
I made a PR in the new repo https://github.com/duneanalytics/dune-v1-abstractions/pull/2

**Comment by jeff-dude** [comment]
closing due to new repo link above

**Pull request closed without merging**